api/server/user/data.go

```diff
@@ -199,8 +199,10 @@ func (s *Service) CidInfo(ctx context.Context, req *userPb.CidInfoRequest) (*use
 			rpcQueudJobs[i] = rpcJob
 		}
 		cidInfo.QueuedStorageJobs = rpcQueudJobs
-		executingJobs := i.ExecutingStorageJobs()
+		executingJobs := i.ExecutingStorageJobs(cid)
 		if len(executingJobs) > 0 {
+			// There is exactly one job in the slice because we specified a cid
+			// and there can be only one executing job per cid at a time.
```

Comment:
May worth doing `if len(executingJobs) == 1` and `log.Warn` in an `else`?
Maybe a bit paranoid, so dismiss depending on your feelings.

Reply:
Yea I like that